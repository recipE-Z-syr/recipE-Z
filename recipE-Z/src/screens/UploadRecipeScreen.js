import React, { Component } from 'react';
import { Button, StyleSheet, Text, TextInput, View, SafeAreaView, Keyboard, TouchableOpacity, Image, Dimensions, ScrollView } from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import defaultStyles from './stylesheet';
import NavigationBar from './NavigationBar';

class PageHeader extends React.Component {
	render () {
		const {title, trianglePosition} = this.props;
		return (
		<View style= {{position: 'absolute', top: 60}} >
		  <Image source={require('../img/header_triangle.png')} style={{width: 15, height: 10, right: trianglePosition}} />
		  <Image source={require('../img/header_rectangle.png')} style={{width: Dimensions.get('window').width, height: 60}} />
		  <Text style={{fontFamily: 'nunito-black', color: '#fff', fontSize: 28, top: -50, left: 15}}>{title}</Text>
		</View>
		);
	}
}

class UploadRecipeScreen extends React.Component {
	constructor(props) {
	super(props)

	this.state = {
	  recipeTitle: '',
	  ingredientsMeas: [],
	  steps: [],
	  time: '',
	  servings: '',
	  difficulty: '',
	  keywords: [],
	  additionalNotes: '',
	  picture: '',
	  addIng: 0,
	  addIngKeys:[],
	  ingredients: {},
	  addStep: 0,
	  addStepKeys: [],
	  steps: {}
	};

	this.addIngredientInput = this.addIngredientInput.bind(this);
	this.addStepInput = this.addStepInput.bind(this);
	}

	addIngredientInput() {
		let inputs = []
		this.state.addIngKeys.forEach( i => {
			inputs.push(
			<View key={'ing'+i} style={{flexDirection: 'row', alignItems: 'center'}} >
			<TextInput
			style={{ width: '92%', height: 40, borderColor: '#ccc', borderWidth: 1, paddingLeft: 10, paddingRight: 10, paddingTop: 5, paddingBottom: 5, marginTop: 8 }}
			onChangeText={text => {
			 let o = {}; o[i] = text;
			 this.setState(
			  (prevState) => ({
			    ingredients: Object.assign({}, prevState.ingredients, o)
			  }));
			}}
			/>
			<TouchableOpacity
			onPress={e => {
				var array = this.state.addIngKeys;
			    var index = array.indexOf(i);
			    if (index !== -1) array.splice(index, 1);
				this.setState({addIngKeys: array});
				let o = {}; o[i] = undefined;
				 this.setState(
				  (prevState) => ({
				    ingredients: Object.assign({}, prevState.ingredients, o)
				  }));
				}
			}>
			<Text style={{ marginLeft: 10, fontSize: 30, fontFamily: 'nunito-extrabold', color: '#ed4848', zIndex: 2}}>x</Text>
			</TouchableOpacity>
			</View>
			);
		});
		return inputs;
	}

	addStepInput() {
		let inputs = []
		this.state.addStepKeys.forEach( i => {
			inputs.push(
			<View key={'ing'+i} style={{flexDirection: 'row', alignItems: 'center'}} >
			<TextInput
			style={{ width: '92%', height: 40, borderColor: '#ccc', borderWidth: 1, paddingLeft: 10, paddingRight: 10, paddingTop: 5, paddingBottom: 5, marginTop: 8 }}
			onChangeText={text => {
			 let o = {}; o[i] = undefined;
			 this.setState(
			  (prevState) => ({
			    steps: Object.assign({}, prevState.steps, o)
			  }));
			}}
			/>
			<TouchableOpacity
			onPress={e => {
				var array = this.state.addStepKeys;
			    var index = array.indexOf(i);
			    if (index !== -1) array.splice(index, 1);
				this.setState({addStepKeys: array})
				}
			}>
			<Text style={{ marginLeft: 10, fontSize: 30, fontFamily: 'nunito-extrabold', color: '#ed4848', zIndex: 2}}>x</Text>
			</TouchableOpacity>
			</View>
			);
		});
		return inputs;
	}

	render() {
	const {navigation} = this.props;
	return (
	  <View style= {[defaultStyles.searchContainer]}>
	  <NavigationBar navigation={this.props.navigation}/>
	  <PageHeader title='Create Recipe' trianglePosition={-272} />
	    <ScrollView style= {{width: '100%', top: 150}} showsVerticalScrollIndicator={false}>
	      <View style={{marginBottom: 20}}>
		      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848', marginBottom: 8}}>Recipe Name</Text>
		      <TextInput
		      style={{ height: 40, borderColor: '#ccc', borderWidth: 1, paddingLeft: 10, paddingRight: 10, paddingTop: 5, paddingBottom: 5 }}
		      onChangeText={text => this.setState({recipeTitle: text})}
		      placeholder='Required'
		      />
	      </View>
	      <View style={{marginBottom: 20}}>
		      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848', marginBottom: 8}}>Ingredients with Measurements</Text>
		      <TextInput
		      style={{ height: 40, borderColor: '#ccc', borderWidth: 1, paddingLeft: 10, paddingRight: 10, paddingTop: 5, paddingBottom: 5 }}
		      placeholder='Required'
		      />
		      {this.addIngredientInput()}
		      <TouchableOpacity style={{flexDirection:'row', alignItems:'center', marginTop: 8}}
		      	onPress = {e => this.setState({addIngKeys: [...this.state.addIngKeys, this.state.addIng], addIng: this.state.addIng +1})}
		      >
		      <Image 
              style={{height: 15, width:15, marginRight: 8}}
              source={require('../img/plus.png')}
              />
		      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848'}}>Click to add another ingredient</Text>
              </TouchableOpacity>
	      </View>
	      <View style={{marginBottom: 20}}>
		      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848', marginBottom: 8}}>Step-by-step Instructions</Text>
		      <TextInput
		      style={{ height: 40, borderColor: '#ccc', borderWidth: 1, paddingLeft: 10, paddingRight: 10, paddingTop: 5, paddingBottom: 5 }}
		      placeholder='Required'
		      />
		      {this.addStepInput()}
		      <TouchableOpacity style={{flexDirection:'row', alignItems:'center', marginTop: 8}}
		      	onPress = {e => this.setState({addStepKeys: [...this.state.addStepKeys, this.state.addStep], addStep: this.state.addStep +1})}
		      >
		      <Image 
              style={{height: 15, width:15, marginRight: 8}}
              source={require('../img/plus.png')}
              />
		      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848'}}>Click to add another step</Text>
              </TouchableOpacity>
	      </View>
	      <View style={{marginBottom: 20}}>
		      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848', marginBottom: 8}}>Total Time</Text>
		      <TextInput
		      style={{ height: 40, borderColor: '#ccc', borderWidth: 1, paddingLeft: 10, paddingRight: 10, paddingTop: 5, paddingBottom: 5 }}
		      onChangeText={text => this.setState({time: text})}
		      placeholder='Optional'
		      />
	      </View>
	      <View style={{marginBottom: 20}}>
		      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848', marginBottom: 8}}>Servings</Text>
		      <TextInput
		      style={{ height: 40, borderColor: '#ccc', borderWidth: 1, paddingLeft: 10, paddingRight: 10, paddingTop: 5, paddingBottom: 5 }}
		      onChangeText={text => this.setState({servings: text})}
		      placeholder='Optional'
		      />
	      </View>
	      <View style={{marginBottom: 20}}>
		      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848', marginBottom: 8}}>Difficulty</Text>
		      <TextInput
		      style={{ height: 40, borderColor: '#ccc', borderWidth: 1, paddingLeft: 10, paddingRight: 10, paddingTop: 5, paddingBottom: 5 }}
		      onChangeText={text => this.setState({difficulty: text})}
		      placeholder='Optional'
		      />
	      </View>
	      <View style={{marginBottom: 20}}>
		      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848', marginBottom: 8}}>Keywords (separate by commas)</Text>
		      <TextInput
		      style={{ height: 40, borderColor: '#ccc', borderWidth: 1, paddingLeft: 10, paddingRight: 10, paddingTop: 5, paddingBottom: 5 }}
		      onChangeText={text => this.setState({keywords: text.split(',').forEach(w => w.trim())})}
		      placeholder='Optional'
		      />
	      </View>
	      <View>
		      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848', marginBottom: 8}}>Additonal Notes</Text>
		      <TextInput
		      multiline
		      style={{ height: 40, borderColor: '#ccc', borderWidth: 1, paddingLeft: 10, paddingRight: 10, paddingTop: 5, paddingBottom: 5, height: 100 }}
		      onChangeText={text => this.setState({additionalNotes: text})}
		      placeholder='Optional'
		      />
	      </View>
	      <Text style={{fontFamily:'nunito-semibold', fontSize:15, color:'#ed4848', marginTop: 20}}>Upload a Picture of Your Dish (Optional)</Text>
	      <TouchableOpacity style={{marginTop: 8, height: 60, width: 60}} >
	      <Image style={{height: 60, width: 60}} source={require('../img/uploadPicture.png')}/>
	      </TouchableOpacity>
	      <TouchableOpacity
	        style={[defaultStyles.redButton, {alignSelf: 'center', marginTop: 30, marginBottom: 200}]}
	        onPress={() => navigation.navigate('Search')}
	        underlayColor='#ed4848'>
	        <Text style={defaultStyles.redButtonText}>SAVE</Text>
	      </TouchableOpacity>
	  	</ScrollView>
	  </View>
	); //end return
 }//end render
}//end class

export default UploadRecipeScreen;
