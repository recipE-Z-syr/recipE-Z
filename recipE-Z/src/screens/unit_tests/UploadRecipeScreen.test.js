//UploadRecipeScreen.test.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, ShallowWrapper } from "enzyme";

import UploadRecipeScreen from '../UploadRecipeScreen';

describe('<UploadRecipeScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<UploadRecipeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should change state if title is entered', () => {
    const instanceOf = renderer.create(<UploadRecipeScreen />).getInstance();
    instanceOf.handleTitle('Some Title');
    expect(instanceOf.state.recipeTitle).toEqual('Some Title');
  });
  it('should change state if time is entered', () => {
    const instanceOf = renderer.create(<UploadRecipeScreen />).getInstance();
    instanceOf.handleTime('3 min');
    expect(instanceOf.state.time).toEqual('3 min');
  });
  it('should change state if servings is entered', () => {
    const instanceOf = renderer.create(<UploadRecipeScreen />).getInstance();
    instanceOf.handleServings('3');
    expect(instanceOf.state.servings).toEqual('3');
  });
  it('should change state if difficulty is entered', () => {
    const instanceOf = renderer.create(<UploadRecipeScreen />).getInstance();
    instanceOf.handleDifficulty('medium');
    expect(instanceOf.state.difficulty).toEqual('medium');
  });
  it('should change state if keywords are entered', () => {
    const instanceOf = renderer.create(<UploadRecipeScreen />).getInstance();
    instanceOf.handleKeywords('tex mex, fresh, easy, tacos');
    expect(instanceOf.state.keywords).toEqual(['tex mex', 'fresh', 'easy', 'tacos']);
  });
  it('should change state if additional notes are entered', () => {
    const instanceOf = renderer.create(<UploadRecipeScreen />).getInstance();
    instanceOf.handleAdditionalNotes('medium');
    expect(instanceOf.state.additionalNotes).toEqual('medium');
  });
});

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  }
});

test('Upload Recipe navigates to Search Screen', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<UploadRecipeScreen {...props} />);  

  const addIngButton = wrapper.find('TouchableOpacity').at(3)
  addIngButton.props().onPress()

  expect(props.navigation.navigate).toBeCalledWith('Search')
})

test('Add another ingredient input', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<UploadRecipeScreen {...props} />);  

  const addStepButton = wrapper.find('TouchableOpacity').at(0)
  addStepButton.props().onPress()

  expect(wrapper.state().addIng).toEqual(1)
})

test('Delete step input', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<UploadRecipeScreen {...props} />);  

  const addIngButton = wrapper.find('TouchableOpacity').at(0)
  addIngButton.props().onPress()
  const deleteButton = wrapper.find('TouchableOpacity').at(0)
  deleteButton.props().onPress()

  expect(wrapper.state().addIng).toEqual(1)
  expect(wrapper.state().addIngKeys).toEqual([])
})

test('Add another step input', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<UploadRecipeScreen {...props} />);  

  const addStepButton = wrapper.find('TouchableOpacity').at(1)
  addStepButton.props().onPress()
  const deleteButton = wrapper.find('TouchableOpacity').at(1)
  deleteButton.props().onPress()

  expect(wrapper.state().addStep).toEqual(1)
  expect(wrapper.state().addStepKeys).toEqual([])
})