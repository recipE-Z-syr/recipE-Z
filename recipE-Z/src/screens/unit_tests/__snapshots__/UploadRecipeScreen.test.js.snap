// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadRecipeScreen /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "flex": 5,
        "justifyContent": "center",
        "paddingLeft": 32,
        "paddingRight": 32,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row-reverse",
        "position": "absolute",
        "right": 30,
        "top": 30,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginLeft": 20,
            "width": 25,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginLeft": 20,
            "width": 25,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginLeft": 20,
            "width": 25,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
        "top": 60,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 10,
          "right": -272,
          "width": 15,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 60,
          "width": 750,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "nunito-black",
          "fontSize": 28,
          "left": 15,
          "top": -50,
        }
      }
    >
      Create Recipe
    </Text>
  </View>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "top": 150,
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-semibold",
              "fontSize": 15,
              "marginBottom": 8,
            }
          }
        >
          Recipe Name
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Required"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 40,
              "paddingBottom": 5,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-semibold",
              "fontSize": 15,
              "marginBottom": 8,
            }
          }
        >
          Ingredients with Measurements
        </Text>
        <TextInput
          allowFontScaling={true}
          placeholder="Required"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 40,
              "paddingBottom": 5,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginTop": 8,
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 15,
                "marginRight": 8,
                "width": 15,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#ed4848",
                "fontFamily": "nunito-semibold",
                "fontSize": 15,
              }
            }
          >
            Click to add another ingredient
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-semibold",
              "fontSize": 15,
              "marginBottom": 8,
            }
          }
        >
          Step-by-step Instructions
        </Text>
        <TextInput
          allowFontScaling={true}
          placeholder="Required"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 40,
              "paddingBottom": 5,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginTop": 8,
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 15,
                "marginRight": 8,
                "width": 15,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#ed4848",
                "fontFamily": "nunito-semibold",
                "fontSize": 15,
              }
            }
          >
            Click to add another step
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-semibold",
              "fontSize": 15,
              "marginBottom": 8,
            }
          }
        >
          Total Time
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Optional"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 40,
              "paddingBottom": 5,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-semibold",
              "fontSize": 15,
              "marginBottom": 8,
            }
          }
        >
          Servings
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Optional"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 40,
              "paddingBottom": 5,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-semibold",
              "fontSize": 15,
              "marginBottom": 8,
            }
          }
        >
          Difficulty
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Optional"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 40,
              "paddingBottom": 5,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-semibold",
              "fontSize": 15,
              "marginBottom": 8,
            }
          }
        >
          Keywords (separate by commas)
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Optional"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 40,
              "paddingBottom": 5,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#ed4848",
              "fontFamily": "nunito-semibold",
              "fontSize": 15,
              "marginBottom": 8,
            }
          }
        >
          Additonal Notes
        </Text>
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Optional"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 100,
              "paddingBottom": 5,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 5,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ed4848",
            "fontFamily": "nunito-semibold",
            "fontSize": 15,
            "marginTop": 20,
          }
        }
      >
        Upload a Picture of Your Dish (Optional)
      </Text>
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 60,
            "marginTop": 8,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 60,
              "width": 60,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#ed4848",
            "borderColor": "#ed4848",
            "borderRadius": 5,
            "borderWidth": 10,
            "justifyContent": "center",
            "marginBottom": 200,
            "marginTop": 30,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "nunito-bold",
              "paddingLeft": 10,
              "paddingRight": 10,
              "textAlign": "center",
            }
          }
        >
          SAVE
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
